### PR TITLE
Python: Remove the memory tied to the kernel. Update kernel examples and notebooks.

### LINKED ISSUE #4557: Python: Use `Service` enum module in all the notebooks instead of boolean checks
Would it make sense to refactor the `Service` enum to a module that can be used by all the notebooks instead of repeated in each of them? I see this being used not just in those 2 notebooks in the future.

_Originally posted by @juliomenendez in https://github.com/microsoft/semantic-kernel/pull/3557#pullrequestreview-1813610810_

`Service` enum reference: #3557 

We need to unify the way we check for which function to run in all the Python notebooks to match notebooks 10 and 11.
Benefiting from the new *services.py* module.

### LINKED ISSUE #4632: Python: Remove memory property from kernel so that the kernel can have multiple "memories"
In the .NET package, the memory property of the kernel was removed because the 1-to-1 relationship was too limiting. Instead, users should use plugins to add additional memory providers.

### PR DESCRIPTION
**Pull request by moonbox3**
### Motivation and Context

As we move toward v1, it is necessary to remove the single kernel memory attribute so that memory connectors can be handled via plugins. This change also refactors the `import_plugin` method name to be similar with dotnet as `import_plugin_from_object`, and introduces a simple `kernel.invoke_prompt` method.

### Description

Updates are:
- The memory connector is now passed to the `SemanticTextMemory` class, along with the embedding generator. This allows the user to add memory to multiple plugins/services as needed. Closes #4632
- Kernel Examples and Notebooks were updated to reflect these changes. Now using Service Enum for all notebooks. Closes #4557 
- There were planners using memory related to the kernel, to store the functions, and many of these planners are non-existent in dotnet's v1 SDK. Python has pending work to update the planner/remove deprecated planners before getting to v1. 
- A Kernel method called `invoke_prompt` was introduced to invoke a simple prompt, and return the `FunctionResult` when complete.
- All unit/integration tests are passing.

### Contribution Checklist

- [X] The code builds clean without any errors or warnings
- [X] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [X] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: python/semantic_kernel/functions/kernel_function.py [hunk 1]
```diff
@@ -177,9 +177,9 @@ def return_parameter(self) -> Optional[KernelParameterMetadata]:
         return self.metadata.return_parameter
 
     @staticmethod
-    def from_native_method(method: Callable[..., Any], plugin_name: str) -> "KernelFunction":
+    def from_method(method: Callable[..., Any], plugin_name: str) -> "KernelFunction":
```
Comment (eavanvalkenburg) [inline python/semantic_kernel/functions/kernel_function.py:180]: please revert this, I will fix in kernelfunction work.

**Review by eavanvalkenburg — APPROVED** [review]
Looks great, two small comments, but non-blocking!

File: python/semantic_kernel/kernel.py [hunk 2]
```diff
@@ -364,6 +354,38 @@ async def invoke(
...
+            Optional[Union[FunctionResult, List[FunctionResult]]]: The result of the function(s)
+        """
+        if not arguments:
+            arguments = KernelArguments(**kwargs)
+        if not prompt:
+            raise AIException(
+                error_code=AIException.ErrorCodes.InvalidPrompt,
+                message="The prompt is either null or empty.",
+            )
+        function = KernelFunction.from_prompt(
```
[the remark this answers is not in the document]
Comment (moonbox3) [reply python/semantic_kernel/kernel.py:383]: It's an interesting idea to store the created function, but I think it kind of defeats the purpose of this slimmer method (or makes this method more complex?). If we store the plugin, we then need to introduce the plugin name/function name as args to the method and add conditionals so the user can use this method to invoke the prompt? Or would the user only use the other methods: `invoke`/`invoke_stream`? Let's leave this simpler method for now, and we can see if we need to make the update to store the function later.

**Review by juliomenendez — APPROVED** [review]
LGTM

**Pull request merged**